llama_index/download/download_utils.py

```diff
@@ -215,8 +224,14 @@ def download_module_and_reqs(
                     f.write(f"from .{module_id} import {', '.join(loader_exports)}")
                     existing_exports = get_exports(f.read())
                 rewrite_exports(existing_exports + loader_exports, str(local_dir_path))
-            with open(f"{module_path}/{extra_file}", "w") as f:
-                f.write(extra_file_raw_content)
+
+            if ".pdf" in extra_file:
+                extra_file_raw_content = cast(bytes, extra_file_raw_content)
```

Comment:
so for pdf, we need to cast to bytes first?

Reply:
it's actually bytes, but we get lint issues — I believe its mypy that's unable to tell the type in this case.